Moodis/Moodis/Feature/Login/LoginViewModel.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Moodis.Feature.Login
+{
+    class LoginViewModel
+    {
+
+        public User Authenticate(String username, String password)
+        {
+            List<User> userList = new List<User>();
+
+            if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin"))
+            {
+                userList = Serializer.Load<List<User>>(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin");
+            }
+
+            if (userList.Exists(x => x.getUsername() == username && x.getPassword() == CalculateMD5Hash(password)))
+            {
+                return userList.Find(x => x.getUsername() == username && x.getPassword() == CalculateMD5Hash(password));
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public bool AddUser(String username, String password)
+        {
+            List<User> userList = new List<User>();
+
+            if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin"))
+            {
+                userList = Serializer.Load<List<User>>(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin");
+            }
+
+            if (userList.Exists(x => x.getUsername() == username))
+            {
+                return false;
+            }
+            else
+            {
+                User user = new User(username, CalculateMD5Hash(password));
+                userList.Add(user);
+                Serializer.Save(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin", userList);
+                return true;
+            }
+        }
+
+        private string CalculateMD5Hash(string input)
```

Comment:
It is great that you used cryptography, but I believe this method should be moved to a different class, like Crypto.cs or something